packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/sync/MutableSubscriptionSet.kt

```diff
@@ -0,0 +1,117 @@
+     * that matches the [query] and the [name], this operation does nothing and the existing
+     * subscription will be returned.
+     *
+     * If an existing named subscription exists on a different query an [IllegalArgumentException]
+     * will be thrown unless [updateExisting] is set to `true`, in which case the existing
+     * subscription will be updated with the new query.
+     *
+     * @param query the query that will be subscribed to. Note, subscription queries have
+     * restrictions compared to normal queries.
+     * @param name the name of the subscription. If no name is provided, the subscription is
+     * considered to be anonymous.
+     * @param updateExisting determines the behaviour if an existing named subscription
+     * already exists. This does nothing for anonymous subscriptions.
+     * @return the newly added subscription.
+     * @throws IllegalArgumentException if a subscription matching the provided one already exists
+     * but on a different query and [updateExisting] was set to `false`.
+     */
+    public fun <T : RealmObject> add(query: RealmQuery<T>, name: String? = null, updateExisting: Boolean = false): Subscription
+
+    /**
+     * Remove a subscription.
+     *
+     * @param subscription subscription to remove
+     * @return `true` if the subscription was removed, `false` if not.
+     */
+    public fun remove(subscription: Subscription): Boolean
+
+    /**
+     * Remove a named subscription.
+     *
+     * @param name name of the subscription to remove.
+     * @return `true` if a matching subscription was removed, `false` if [Subscription] could be
+     * found.
+     */
+    public fun remove(name: String): Boolean
+
+    /**
+     * Remove all subscriptions with queries on a given [Subscription.objectType].
+     *
+     * @param objectType subscriptions on this object type will be removed.
+     * @return `true` if one or more subscriptions were removed, `false` if no
+     * subscriptions were removed.
+     */
+    public fun removeAll(objectType: String): Boolean
+
+    /**
+     * Remove all subscriptions with queries on a given given model class.
+     *
+     * @param type subscriptions on this type will be removed.
+     * @return `true` if one or more subscriptions were removed, `false` if no
+     * subscriptions were removed.
+     */
+    public fun <T : RealmObject> removeAll(type: KClass<T>): Boolean
+
+    /**
+     * Remove all subscriptions in this subscription set.
+     *
+     * @return `true` if one or more subscriptions were removed, `true` if the subscription set
+     * was empty.
+     */
+    public fun removeAll(): Boolean
+
+    /**
+     * Creates an anonymous [Subscription] in the current [MutableSubscriptionSet] directly from
+     * a [RealmQuery].
+     *
+     * @return the [Subscription] that was added.
+     */
+    public fun RealmQuery<out RealmObject>.subscribe(): Subscription = add(this)
```

Comment:
I was just about ordering these methods that create subscriptions together with the `add` methods. So that you have all the creating methods beside each other and all the remove methods below.